### PR TITLE
Packaging

### PR DESCRIPTION
**Pull request by KennethNielsen**
This PR includes the changes needed to make a good `pyproject.toml` based packaging solution. It involves clearing up a few thing and fixing the rest.

1) Removes setup.py.

The move to pyproject.toml was prompted by the idea that setup.py has always almost exclusively been boiler plate, and basically just fed data into the same process. With the new approach, we enter that data into the toml file and use a build backend to build the packages. Therefore, the information that used to be in setup.py is redundant and the file has been removed.

2) Separates development requirements from pyproject.toml

This fixes a misunderstanding of mine (though I can see that the tool is often being used wrong in this way). pyproject.toml is meant **only** for tool configuration and package setup **not** for development dependencies. Having them in there, actually means that they become optional dependencies of _the package_, which is never what was intended. These dependencies are moved back out into requriements files, although the primary package dependencies still lives in the toml file where they belong. This move had the 2 additional benefits of; getting the source dependency (which the packaging backends complains about) out of pyproject.toml and making it easier to work with the source dependency in RTD, where we have to take special measures to get it installed.

3) A build task has been added, so that we don't need to remember commands.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -26,36 +24,44 @@ dependencies = [
...
 [tool.setuptools.dynamic]
 version = {attr = "spectro_inlets_quantification.__version__"}
-# readme = {file = ["README.md"]}
-# The readme prevents uploading because PyPI expects rst. I specified in setup.py that it is
-#   markdown, but can't find the syntax to do it here. See:
-#   https://packaging.python.org/en/latest/specifications/core-metadata/#description-content-type
+# readme = {file = ["README.md"], content-type = "text/markdown"}
+
+[tool.setuptools.package-data]
+your_package_name = [
```
Comment (ScottSoren) [inline pyproject.toml:54]: This is fine, but isn't there a way to include any .yml file in data, however many subfolders deep it is?

File: requirements-doc.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# We depend on an unreleased feature of sphinx-proof for building the docs. Installing directly
```
Comment (ScottSoren) [inline requirements-doc.txt:1]: Nice with explanation.

File: src/spectro_inlets_quantification/chip.py [hunk 3]
```diff
@@ -99,8 +99,10 @@ def save(self, file_name: str, chip_dir: Optional[PATHLIKE] = None, **kwargs: An
...
                 file_name.endswith(".yml")
-            chip_dir: path to directory to save chip in, defaults to :attr:`Config.chip_directory`
-                the spitze chips folder.
+            chip_dir: path to directory to save chip in. Defaults to
+                ``:attr:`config.aux_data_directory` / "chips"`` if the
+                `aux_data_directory` is set and if not defaults to
+                ``:attr:`config.data_directory` / "chips"``
             kwargs: (other) key word arguments are added to self_as_dict before saving.
         """
         file_name_with_suffix = Path(file_name).with_suffix(".yml")
```
Comment (ScottSoren) [inline src/spectro_inlets_quantification/chip.py:108]: Should this also have the `Path(self.name + ".yml")` default?

**Review by ScottSoren — APPROVED** [review]
Looks good!
The only question (it may be done right) is whether one can save a `SignalProcessor` or a `Chip` without specifying a `file_name` as can now be done for `Molecule` and `Calibration`.

File: pyproject.toml (hunk 1 above)
Comment (KennethNielsen) [reply pyproject.toml:54]: Yes there is, good catch

File: src/spectro_inlets_quantification/chip.py (hunk 3 above)
Comment (KennethNielsen) [reply src/spectro_inlets_quantification/chip.py:108]: I don't think it can. The `Chip` doesn't have a name

Comment (ScottSoren) [reply src/spectro_inlets_quantification/chip.py:108]: Wait, but shouldn't it? The name would typically be the serial number, presumably.

Comment (ScottSoren) [reply src/spectro_inlets_quantification/chip.py:108]: It doesn't have to do with packaging, though. I've made an issue of it so we can merge this.
https://github.com/SpectroInlets/spectro_inlets_quantification/issues/6

**Review by ScottSoren — APPROVED** [review]

**Pull request merged**